magenta/music/musicxml_parser.py

```diff
@@ -293,37 +291,34 @@ class ScorePart(object):
   A <score-part> element contains MIDI program and channel info
   for the <part> elements in the MusicXML document.
 
-  If no MIDI info is found for the part, use the next available
-  MIDI channel and default to the Grand Piano program (MIDI Program #1)
+  If no MIDI info is found for the part, use the default MIDI channel (0)
+  and default to the Grand Piano program (MIDI Program #1).
   """
 
-  def __init__(self, state, xml_score_part=None):
-    self.xml_score_part = xml_score_part
+  def __init__(self, xml_score_part=None):
+    self.id = ''
     self.part_name = ''
     self.midi_channel = DEFAULT_MIDI_CHANNEL
     self.midi_program = DEFAULT_MIDI_PROGRAM
-    self.state = state
     if xml_score_part is not None:
-      self._parse()
+      self._parse(xml_score_part)
 
-  def _parse(self):
+  def _parse(self, xml_score_part):
     """Parse the <score-part> element to an in-memory representation."""
-    if self.xml_score_part.find('part-name') is not None:
-      self.part_name = self.xml_score_part.find('part-name').text
+    self.id = xml_score_part.attrib['id']
```

Comment:
Yes, it is required by the MusicXML specification: http://usermanuals.musicxml.com/MusicXML/MusicXML.htm#EL-MusicXML-part.htm